### ISSUE TITLE
Add SSH deploy keys

### ISSUE THREAD

**Issue by stonewareslord** [gogits/gogs#3880]
From https://discuss.gogs.io/t/ssh-push-deploy-keys/675

I think it would be helpful to have deploy keys that can have different permissions. For example, I could make one deploy key read only and another able to push/pull. One example use case is a Jenkins server pushing compiled website files daily. 

Instead of adding a key attached to your Gogs account, you can make a temporary key that is project or organization specific. This way, the key can be added to Jenkins and everyone who has access to the key wouldn't have access to your personal projects.

**Comment by ogarcia** [comment]
I think that this feature can be as in GitHub

![keys](https://cloud.githubusercontent.com/assets/572924/22512389/24813eac-e899-11e6-966d-e50718804759.png)

A check that allow write/push access to a deploy key :wink:

**Comment by tobischulz** [comment]
+1 for deploy keys with write access

**Comment by andrejsavikin** [comment]
Yes, this feature is avaliable on github too, and it's amazing how useful it is

**Comment by gilliek** [comment]
+1

**Comment by stonewareslord** [comment]
For those still following this issue, Gitea (an amazing fork of Gogs) has this feature and many many others. I’ve since starting using it instead of Gogs

**Comment by ties** [comment]
+1

**Comment by stonewareslord** [comment]
It seems my comment was deleted. For those looking into this functionality, you can find it in Gitea. 

This looks to be the commit that added it: https://github.com/go-gitea/gitea/commit/e78786ef39526fac6705a831fc291c55f1ca091b

**Comment by mikeperalta1** [comment]
+1 this would be great

**Comment by dasginganinja** [comment]
+1

**Comment by junes** [comment]
+1
